server.py

```diff
@@ -1,182 +1,273 @@
-
-    if not cart:
-        return jsonify("Cart with id '{}' was not found.".format(uid)), status.HTTP_404_NOT_FOUND
-
-    try:
-        cart.add_products(request.get_json())
-        cart.save()
-    except DataValidationError as e:
-        message = { 'error': e.args[0] }
-        return jsonify(message), status.HTTP_400_BAD_REQUEST
-
-    return make_response(jsonify(cart.serialize()), status.HTTP_200_OK)
-
-######################################################################
-# DELETE A PRODUCT FROM A SHOPCART
-######################################################################
-@app.route('/shopcarts/<int:uid>/products/<int:pid>', methods=['DELETE'])
-def delete_product(uid, pid):
-    cart = Shopcart.find(uid)
-    if cart:
-        cart.delete_product(pid)
-    return make_response('', status.HTTP_204_NO_CONTENT)
-
-
-
-######################################################################
-#  U T I L I T Y   F U N C T I O N S
-######################################################################
-
-def check_content_type(content_type):
-    """ Checks that the media type is correct """
-    if request.headers['Content-Type'] == content_type:
-        return
-    #app.logger.error('Invalid Content-Type: %s', request.headers['Content-Type'])
-    abort(status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, 'Content-Type must be {}'.format(content_type))
-
-
-
-######################################################################
-# List all Shopcarts
-######################################################################
-@app.route('/shopcarts', methods=['GET'])
-def get_all_shopcarts():
-    pid = request.args.get('pid')
-    if pid:
-        carts = Shopcart.find_by_product( int(pid) )
-    else:
-        carts = Shopcart.all()
-    message = [cart.serialize() for cart in carts]
-    rc = status.HTTP_200_OK
-    return jsonify(message), rc
-
-######################################################################
-# Prune empty Shopcarts
-######################################################################
-@app.route('/shopcarts/prune', methods=['DELETE'])
-def prune_empty_shopcarts():
-    Shopcart.prune()
-    return make_response('', status.HTTP_204_NO_CONTENT)
-
-if __name__ == "__main__":
-    # dummy data for server testing
-    app.run(host='0.0.0.0', port=int(PORT), debug=DEBUG)
+import os
+import json
+from flask import Flask, jsonify, request, url_for, make_response, abort
+from flask_api import status
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import create_engine
+
+from models.shopcart import Shopcart
+from models.dataerror import DataValidationError
+
+# Local database credentials
+HOSTNAME = '127.0.0.1'
+PORT = '3306'
+DATABASE = 'localdb'
+USERNAME = 'root'
+PASSWORD = 'root'
+DB_URI = 'mysql+pymysql://root:root@localhost/localdb'
+
+# Get service credentials from Bluemix
+if 'VCAP_SERVICES' in os.environ: 
+    vcap = json.loads(os.environ['VCAP_SERVICES'])
+    creds = vcap['rediscloud'][0]['credentials']

```

Comment:
You're still reading from `rediscloud` here, when we're using a MySQL/ClearDB database - the bluemix site has the credentials, so you can check the structure of the JSON object

Reply:
You are right. I can fix it.